onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -261,6 +509,83 @@ class ReduceAggregatorMin : public ReduceAggregator<T, TVAL> {
     return Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>>(from_data, this->N_).minCoeff();
   }
   inline void update(const T& v) { this->accumulator_ = v < this->accumulator_ ? v : this->accumulator_; }
+
+  // Fast reduction
+  static inline FastReduceKind fast_reduce() { return FastReduceKindValues::KR | FastReduceKindValues::RK | FastReduceKindValues::KRK; }
+
+  static void FastReduceKR(const Tensor& input, const std::vector<int64_t>& fast_shape,
+                           Tensor& output, concurrency::ThreadPool* tp) {
+    ORT_ENFORCE(fast_shape.size() == 2, "Only works on matrices with two dimensions.");
+    ORT_ENFORCE(fast_shape[0] == output.Shape().Size(), "Output size mismatch.");
+    if (fast_shape[0] >= 4) {
+      const T* data = input.Data<T>();
+      T* out = output.MutableData<T>();
+      int64_t stridei = fast_shape[1];
+      concurrency::ThreadPool::TryBatchParallelFor(
+          tp,
+          SafeInt<int32_t>(fast_shape[0]),
+          [data, stridei, out](ptrdiff_t j) {
+            out[j] = ConstEigenVectorArrayMap<T>(data + j * stridei, stridei).minCoeff();
+          },
+          0);
+    } else {
+      EigenVectorMap<T>(output.MutableData<T>(), fast_shape[0]) = ConstEigenMatrixMap<T>(input.Data<T>(), fast_shape[1], fast_shape[0]).colwise().minCoeff();
+    }
```

Comment:
nit: please keep lines to 120 chars as per coding standards

[https://marketplace.visualstudio.com/items?itemName=PaulHarrington.EditorGuidelines](https://marketplace.visualstudio.com/items?itemName=PaulHarrington.EditorGuidelines) is very handy at providing a visual line.

Reply:
Sorry for that, I forgot to check, I though there was a unit test failing if a style rule was not followed. I also wonder which tool you used to notice that since I don't see any indication of the line size in the code review.

Reply:
Line just looked really long and I manually checked it. We don't currently have a check that enforces the line length. 

Enforcing it via clang-format results in horrible code as it tries to fill as much of the line as possible up to the limit, so any line breaks a developer adds to try and break things up in a meaningful way get lost. We may look at adding a CI that checks clang-format has been applied. That could also check line lengths via a simple python script if there's no better option.

Reply:
I checked some other lines too.